packages/e2e.test.foundry-sdk-generator/generateMockOntology.js

```diff
@@ -14,21 +14,29 @@
  * limitations under the License.
  */
 
+// @ts-check
+import { __testSeamOnly_NotSemverStable__GeneratePackageCommand as GeneratePackageCommand } from "@osdk/foundry-sdk-generator";
 import { apiServer } from "@osdk/shared.test";
-import { rm } from "fs/promises";
-import { join } from "path";
-import { GeneratePackageCommand } from "../generate/index.js";
+import * as fs from "node:fs/promises";
+import * as path from "node:path";
+import { fileURLToPath } from "node:url";
 
-const dir = `${__dirname}/../generatedNoCheck/`;
-export async function setup() {
+const dir = path.join(
+  path.dirname(fileURLToPath(import.meta.url)),
+  "src",
+  "generatedNoCheck",
+);
+async function setup() {
   apiServer.listen();
 
-  try {
-    await rm(join(dir, "@test-app"), { recursive: true });
-    await rm(join(dir, "@test-app2"), { recursive: true });
-  } catch (e) {
-    // Only needed for regenerations
-  }
+  const testAppDir = path.join(dir, "@test-app");
+  const testApp2Dir = path.join(dir, "@test-app2");
+
+  await rmRf(testAppDir);
+  await rmRf(testApp2Dir);
+
+  await safeStat(testAppDir, "should not exist");
+  await safeStat(testApp2Dir, "should not exist");
```

Comment:
After upgrading to vitest 2, on initial runs of this test suite, everything would fail due to not being able to find the generated code. And then later when I was testing to find out what was happening I realized that if you bailed on `setup` early (after deleting) that it ALSO was failing. So I added some safety checks to be sure that things were being deleted (they were).

Turns out that in vitest 1, the global setup would run before any forks for the test runners would happen but in vitest 2, the forks happen right away. There also seemed to be some caching of the filesystem on initial load such that no matter what, on the first run, even though I could confirm we fully regenerated prior to the test running, the test would still think those files didn't exist.

I figured there was no harm in keeping the extra sanity checks after I chased the behavior down, so it remains.